### DISCUSSION TITLE
Errors and warnings with Quarto

### DISCUSSION THREAD

**Discussion by asadow** [ropensci/tarchetypes#143] [Help]
With Quarto 1.3.333 and Windows 11,

I can use `tar_quarto_rep()` fine provided 2 things:

1) the input files live at the project root
2) `_quarto.yml` does not have an `output-dir:` set (e.g. to `_outputs`)

If I don't follow 1, I get,
```
Error:
! Error running targets::tar_make()
  Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
  Debugging guide: https://books.ropensci.org/targets/debugging.html
  How to ask for help: https://books.ropensci.org/targets/help.html
  Last error: ! ! System command 'quarto.exe' failed
```
Maybe this somehow relates to [this issue](https://github.com/quarto-dev/quarto-r/issues/99)?

If I don't follow 2, I get

```
Warning messages:
1: cannot rename file 'file26cc44e64f24.html' to 'my-file.html', reason 'The system cannot find the file specified' 
```

(As an aside, I noticed tar_make() will still skip the reports if `output-dir:` in `_quarto.yml` has changed. Not sure if that is of concern.)

**Comment by wlandau** [comment]
@asadow, would you try with development `tarchetypes`?

```r
remotes::install_github("ropensci/tarchetypes")
```

I thought I fixed this issue via #144 (migrated from the `targets` issue tracker).

**Reply by asadow** [reply]
After trying with the dev version, I still get the warning if I don't follow 2. 

But now I don't get the error if 1 is not followed, where actually I should (or a warning) because under the hood these warnings happen if I individually quarto render a file:

> [WARNING] Could not fetch resource _template_files/libs/clipboard/clipboard.min.js
> [WARNING] Could not fetch resource _template_files/libs/quarto-html/quarto.js
> ....
> 

This is related to [this discussion](https://github.com/quarto-dev/quarto-cli/discussions/4041). And I see you've also posted on [whether output-file should take a path](https://github.com/quarto-dev/quarto-cli/issues/2440).

For now I guess I'll resort to creating the files at the root folder and then moving them. Seems strange as I feel Quarto is heralded as a Rmarkdown replacement.

Lastly, is it of concern that tar_make() will still skip the reports if `output-dir:` in _quarto.yml has changed?

**Reply by wlandau** [reply]
> Lastly, is it of concern that tar_make() will still skip the reports if output-dir: in _quarto.yml has changed?

Please see https://github.com/ropensci/tarchetypes/discussions/107. `tar_quarto_rep()` operates on a single Quarto source file as opposed to a project, so I am not sure `_quarto.yml` is expected here.
